ratatui-core/src/buffer/buffer.rs

```diff
@@ -655,6 +661,47 @@ mod tests {
         assert_eq!(result, expected);
     }
 
+    #[test]
+    fn chinese_characters_rendering() {
```

Comment:
I checked this test against the current code and it succeeds already.

Reply:
I’ve added more tests locally and found that the current main branch code will fail when executing these tests. I’ll submit the modifications after conducting more tests and pinpointing some issues.